src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -63,6 +63,20 @@ internal class CSharpSuggestionModeCompletionProvider : SuggestionModeCompletion
                 {
                     return CreateSuggestionModeItem(CSharpFeaturesResources.NamespaceName, itemSpan, CSharpFeaturesResources.AutoselectDisabledDueToNamespaceDeclaration);
                 }
+                else if (tree.IsPartialTypeDeclarationNameContext(position, cancellationToken, out typeDeclaration))
+                {
+                    switch (typeDeclaration.Keyword.Kind())
+                    {
+                        case SyntaxKind.ClassKeyword:
+                            return CreateSuggestionModeItem(CSharpFeaturesResources.ClassName, itemSpan, CSharpFeaturesResources.AutoselectDisabledDueToClassDeclaration);
```

Comment:
As above, I think it's nice to keep attention to detail like this. That said, I don't feel strongly about it, so if there's some hidden cost, I happily trim it back.